pkg/ksym/ksym.go

```diff
@@ -52,8 +57,12 @@ func NewKsymCache(logger log.Logger, reg prometheus.Registerer, f ...fs.FS) *cac
 		fs = f[0]
 	}
 	return &cache{
-		logger:         logger,
-		fs:             fs,
+		logger: logger,
+		fs:     fs,
+		// My machine has ~200k loaded symbols. Reserving 50k entries, as there might be
+		// boxes with fewer symbols loaded, and if we need to reallocate, we would have
+		// to do it once or twice, so this value seems like a reasonable middle ground.
+		kernelSymbols:  make([]ksym, 0, 50000),
```

Comment:
What's the amount of memory pressure that this one puts in the worst-case?

Reply:
Each Ksym is 24B in size + the string overhead, a back-of-the-envelop calculation would be something like:

`50000  (number of ksyms) * (24B (size of the address and string metadata) + 38 characters (P90 length symbols in my box) * 8Bytes/character` = ~ 16400000Bytes = ~16MB 

This is some rough calculation, but I think it should be representative.

I've also added a unittest to ensure we don't add extra fields to the ksym structure without noticing: https://github.com/parca-dev/parca-agent/pull/708/files#diff-694cd04c70fb74c102d9e61bc2c373a0d8df849095e5ddc439af8f2cb5372f40R28

Reply:
Let's add this as a comment as well. I'm fine to add this because we spent a lot of time checking the kernel symbols. If I'm not mistaken, it should've been mentioned in #393.

But also we have #644 to reduce the memory. So let's add this as a comment and keep an eye on this one. I'm sure we have many other things to optimise until it comes to this.

Reply:
Sounds good. Let's make sure that we have the right data and we can always optimise things down the line.


By the way, not sure if you saw the comment about the kernel symbol cache. I am worried about it growing in size a lot as it's an unbounded map. I am planning to add a bounded LRU cache there in another PR.

Reply:
Just go for it. IT would be nice if you can create issues for the things that you discovered.